packages/keyring-eth-ledger-bridge/src/ledger-iframe-bridge.ts

```diff
@@ -169,13 +186,12 @@ export class LedgerIframeBridge
   }
 
   async updateTransportMethod(transportType: string): Promise<boolean> {
-    return new Promise((resolve, reject) => {
-      // If the iframe isn't loaded yet, let's store the desired transportType value and
-      // optimistically return a successful promise
-      if (!this.iframeLoaded) {
-        throw new Error('The iframe is not loaded yet');
-      }
+    // If the iframe isn't loaded yet, the initialization is reattempted
+    if (!this.iframeLoaded) {
+      await this.#init();
```

Comment:
Hmm, we certainly wouldn't want to trigger the creation of the iframe multiple times. Perhaps we could store a pending initialization as a promise somewhere, to prevent concurrent initialization attempts?

Reply:
Good point. I added a deferred promise in https://github.com/MetaMask/accounts/pull/151/commits/fb00879dd8ffdadac8b524515c799a31ab449dba